### PR TITLE
chore(schema): for user-avatar component

### PR DESCRIPTION
**Pull request by yqrashawn**
### Summary

add schema for user-avatar component

status: ready

### CODE REVIEW

File: src/quo/components/avatars/user_avatar/view.cljs [hunk 1]
```diff
@@ -55,10 +57,10 @@
            status-indicator?   true
            online?             true
            ring?               true
-           customization-color :blue}
+           customization-color :blue
+           full-name           "Your Name"}
     :as   props}]
-  (let [full-name          (or full-name "Your Name")
```
Comment (clauxx) [inline src/quo/components/avatars/user_avatar/view.cljs]: `:or` will not handle the case where `full-name` is passed as nil (e.g. used with `(when ...)`). I guess would be better to keep it as was, wdyt?

File: src/quo/components/avatars/user_avatar/view.cljs [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
     [quo.components.avatars.user-avatar.style :as style]
     [quo.components.common.no-flicker-image :as no-flicker-image]
     [quo.components.markdown.text :as text]
     [quo.theme]
     [react-native.core :as rn]
     [react-native.fast-image :as fast-image]
+    [schema.core :as schema]
     utils.string))
 
 (defn initials-avatar
```
Comment (clauxx) [inline src/quo/components/avatars/user_avatar/view.cljs:13]: I think we can remove the docstring with the addition of the schema

File: src/quo/components/avatars/user_avatar/schema.cljs [hunk 3]
```diff
@@ -0,0 +1,26 @@
+(ns quo.components.avatars.user-avatar.schema
+  (:require
+    [quo.components.avatars.user-avatar.style :as style]))
+
```
Comment (clauxx) [inline src/quo/components/avatars/user_avatar/schema.cljs:4]: extra space

File: src/quo/components/avatars/user_avatar/schema.cljs [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+(def ?schema
+  [:=>
+   [:catn
+    [:props
+     [:map
+      [:full-name {:optional true} [:maybe string?]]
+      [:size {:optional true} [:maybe (into [:enum] (keys style/sizes))]]
+      [:customization-color [:maybe :schema.common/customization-color]]
```
Comment (clauxx) [inline src/quo/components/avatars/user_avatar/schema.cljs]: can't it be optional as well?

File: src/quo/components/avatars/user_avatar/schema.cljs [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+      [:static? {:optional true} [:maybe :any]]
+      [:status-indicator? {:optional true} [:maybe :any]]
+      [:online? {:optional true} [:maybe :any]]
+      [:ring? {:optional true} [:maybe :any]]
```
Comment (clauxx) [inline src/quo/components/avatars/user_avatar/schema.cljs]: would these be booleans given the boolean-sounding names, or are they used differently?

File: src/quo/components/avatars/user_avatar/schema.cljs [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+      [:theme :schema.common/theme]
+      [:profile-picture
+       [:maybe
+        [:or
+         string?
+         number?
+         [:map [:uri string?]]
+         [:map [:fn fn?]]]]]]]]
```
Comment (clauxx) [inline src/quo/components/avatars/user_avatar/schema.cljs]: we have a schema for this already `:schema.common/image-source`

**Review by clauxx — CHANGES_REQUESTED** [review]

**Comment by yqrashawn** [comment]
Thanks for the review @clauxx 
fixed above feedback

the doc string is mostly about the `profile-picture-fn` I think we should keep them

File: src/quo/components/avatars/user_avatar/view.cljs (hunk 2 above)
Comment (J-Son89) [reply src/quo/components/avatars/user_avatar/view.cljs:13]: yes, that is something we should check for too. We should just go and nuke majority of doc strings unless there is specific notes/comments in there 👍

**Review by clauxx — APPROVED** [review]
Looks good 👍

**Review by ilmotta — APPROVED** [review]

**Comment by yqrashawn** [comment]
Fix component tests

several `h/render` to `h/render-with-theme-provider`

and

```diff
-(defonce mock-picture {:uri (js/require "../resources/images/mock2/user_picture_male4.png")})
+(defonce mock-picture (resources/mock-images :user-picture-male5))
```

the old one is `{:uri 4}`

status-im.common.resources is not allowed in quo, replaced `(resources/mock-images :user-picture-male5)` with its value `1`

**Pull request merged**